pkg/virt-handler/vm.go

```diff
@@ -2615,6 +2669,32 @@ func (d *VirtualMachineController) hotplugSriovInterfacesCommand(vmi *v1.Virtual
 	return nil
 }
 
+func memoryDumpPath(volumeStatus v1.VolumeStatus) string {
+	target := hotplugdisk.GetVolumeMountDir(volumeStatus.Name)
+	dumpPath := filepath.Join(target, volumeStatus.MemoryDumpVolume.TargetFileName)
+	return dumpPath
+}
+
+func (d *VirtualMachineController) getMemoryDump(vmi *v1.VirtualMachineInstance, volumeStatus v1.VolumeStatus) (bool, error) {
```

Comment:
This function is a little deceiving. It is both invoking the memory dump and checking to see if it has completed.

The pattern we typically use in this controller is that some action is taken in the `sync` logic, and that we observe that action has completed when updating the status.

Reply:
@davidvossel @ShellyKa13 

If I'm understanding correctly, basically this [data](https://github.com/ShellyKa13/kubevirt/blob/memory-dump/pkg/virt-launcher/virtwrap/manager.go#L166-L170) should be moved into the libvirt domain spec metadata (like MigrationMetadata).  During `sync` the metadata is updated and `update status` just looks up the data in the domain

Reply:
>  During sync the metadata is updated and update status just looks up the data in the domain

the metadata would technically be updated by an async goroutine in virt-launcher and feed back to virt-handler via the domain informer. But like you said, update status would just look at the domain from the cache and see whatever metadata fields indicate the memory dump completed are set.